pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -567,7 +567,10 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			cdLog.WithError(err).Log(controllerutils.LogLevel(err), "error adding finalizer")
 			return reconcile.Result{}, err
 		}
-		metricClustersCreated.WithLabelValues(hivemetrics.GetClusterDeploymentType(cd)).Inc()
+		metricClustersCreated.WithLabelValues(hivemetrics.GetClusterDeploymentType(cd, hivev1.HiveClusterTypeLabel),
+			hivemetrics.GetClusterDeploymentType(cd, constants.STSClusterLabel),
+			hivemetrics.GetClusterDeploymentType(cd, constants.PrivateLinkClusterLabel),
+			hivemetrics.GetClusterDeploymentType(cd, constants.ManagedVPCLabel)).Inc()
```

Comment:
As mentioned elsewhere, once we get into more than one label, we should really start using `With()` rather than relying on the label order via `WithLabelValues()`.

Also as discussed, it would be cool to DRY the repetitious currying into a common function. (That way if, say, we needed to change how the label values are computed, we could do it in one place :) )